examples/petsc/bp3.h

```diff
@@ -75,33 +75,33 @@ CEED_QFUNCTION(SetupDiff)(void *ctx, const CeedInt Q,
 
 CEED_QFUNCTION(Diff)(void *ctx, const CeedInt Q,
                      const CeedScalar *const *in, CeedScalar *const *out) {
-  const CeedScalar *ug = in[0], *qd = in[1];
-  CeedScalar *vg = out[0];
+  const CeedScalar *restrict ug = in[0], *restrict qd = in[1];
```

Comment:
@jedbrown do we want to keep these `restrict` qualifiers around, even though you've seen they are ineffective? If that is the case, should I homogenize it for all other examples?

Reply:
Let's remove them since C++ and CUDA (https://devblogs.nvidia.com/cuda-pro-tip-optimize-pointer-aliasing/) use `__restrict__` and it seems to be unnecessary with these pragmas.

Reply:
Removed them.